### PR TITLE
Initial training procedure

### PR DESCRIPTION
**Pull request by stefanDeveloper**
For the first release, we want to support RandomForest, and XGBoost.

### CODE REVIEW

**Comment by stefanDeveloper** [comment]
@lamr02n Please note that the test of the detector is under implementation. Meanwhile please review the remaining code so far.

File: src/train/dataset.py [hunk 1]
```diff
@@ -66,111 +85,326 @@ def preprocess(x: pl.DataFrame):
...
-    @property
-    def bambenek_dataset(self) -> Dataset:
-        return self.bambenek_data
-
-    @property
-    def cic_dataset(self) -> Dataset:
-        return self.cic_data
+    def dgarchive_dataset(self) -> Dataset:
+        self.dgarchive_data = Dataset(
+            data_path=[
```
Comment (lamr02n) [inline src/train/dataset.py:297]: Maybe we should move the paths to their own file to simplify and shorten the code

**Review by lamr02n — APPROVED** [review]
Looks good!

**Review by lamr02n — APPROVED** [review]
LGTM

**Pull request merged**